src/App.tsx

```diff
@@ -1,14 +1,57 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [isLoadingUser, setIsLoadingUser] = useState(false);
+  const [isLoadingTodo, setIsLoadingTodo] = useState(true);
+  const [isOpenTodo, setIsOpenTodo] = useState(0);
+  const [qwery, setQwery] = useState('');
+  const [category, setCategory] = useState('all');
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    getTodos().then((loadedTodos) => {
+      setTodos(loadedTodos);
+      setIsLoadingTodo(false);
+    });
+  }, []);
+
+  const onOpenTodo = (todoId: number) => {
+    setIsOpenTodo(todoId);
+    setIsLoadingUser(true);
+  };
+
+  const filterTodos = () => {
+    const byQwery = todos.filter(todo => {
+      const title = todo.title.toLowerCase();
+
+      return title.match(qwery.toLowerCase());
+    });
+
+    if (category === 'completed') {
+      return byQwery.filter(todo => (todo.completed));
+    }
+
+    if (category === 'active') {
+      return byQwery.filter(todo => (!todo.completed));
+    }
+
+    return byQwery;
+  };
```

Comment:
would be great to implement `debounce` for the todos search
It isn't a required task but I believe in you, you can do it)